src/init.cpp

```diff
@@ -1853,6 +1873,22 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     connOptions.m_i2p_accept_incoming = args.GetBoolArg("-i2pacceptincoming", DEFAULT_I2P_ACCEPT_INCOMING);
 
+    if (args.GetBoolArg("-sensitiverelayowntx", DEFAULT_SENSITIVE_RELAY_OWN_TX)) {
+        // If -listenonion is set, then NET_ONION may not be reachable now
+        // but may become reachable later, thus only error here if it is not
+        // reachable and will not become reachable for sure.
+        if (!IsReachable(NET_I2P) && !IsReachable(NET_ONION) && !listenonion) {
```

Comment:
_moving discussion from https://github.com/bitcoin/bitcoin/pull/27509#issuecomment-1659008138_

> 1. `-sensitiverelay=1` and `-onlynet=ipv4` should probably throw an init error. In this configuration, sensitive stuff seems to be ignored and new wallet TXs are relayed via ipv4 peers like normal.

This check was supposed to prevent that but it was too strict. Apparently you have `listenonion=1` which means that later, after startup, we will connect to the Tor control and may set the Tor proxy and make the Tor network reachable. It is a bit weird that `listenonion` remains with the default `1` when `onlynet=ipv4` is used. It means the node will only make outgoing connections to IPv4, but will listen for and accept incoming connections from the Tor network :eyes:.

I relaxed the check to also trigger the error if `onlynet` has been used (and if `!IsReachable(NET_ONION)` at the time of the check, that means `onlynet` was used but tor was omitted from it).

> Is "only use tor for sensitive relay" a possible configuration?

Yes, if the node does not have I2P connectivity. Then only Tor will be used. But if the node has both Tor and I2P connectivity then both will be used. Do you mean that if both networks are reachable, to use only Tor for sensitive relay? I didn't do that because I see no usecase for it.

Reply:
I meant, if IPv4 and Tor are the only reachable networks, that Tor connections would *only* be made for sensitive relay connections. That's why I tried `-onlynet=ipv4 -sensitiverelayowntx=1`

Reply:
No, I did not consider that.